cardano-simple/src/Cardano/Simple/PlutusLedgerApi/V1/Scripts.hs

```diff
@@ -78,6 +83,32 @@ mkStakeValidatorScriptPlutarch :: Config -> ClosedTerm a -> Either Text StakeVal
 mkStakeValidatorScriptPlutarch conf term =
   StakeValidator . Script . Plutarch.unScript <$> compile conf term
 
+mkPlutarchTypedScript :: Config -> ClosedTerm a -> Either Text (PlutarchTypedScript a)
+mkPlutarchTypedScript config term =
+  PlutarchTypedScript . Script . Plutarch.unScript <$> Plutarch.compile config term
+
+applyPlutarchTypedScript ::
+  Config ->
+  PlutarchTypedScript (a :--> b) ->
+  ClosedTerm a ->
+  Either Text (PlutarchTypedScript b)
+applyPlutarchTypedScript
+  config
+  (PlutarchTypedScript (Script (UPLC.Program _ v1 t1)))
+  term = do
+    (Plutarch.Script (UPLC.Program _ v2 t2)) <- Plutarch.compile config term
+
+    when (v1 GHC./= v2) $ throwError "Script versions differ"
+
+    pure $
+      PlutarchTypedScript $
+        Script $
+          UPLC.Program () v1 $
+            UPLC.Apply () t1 t2
```

Comment:
This looks simple enough, would be happier if you'd extract this in a separate function and make it public (we need applyArgs often)

Reply:
Well, UPLC.Apply () is already exposed by Plutus library :)